ARGB wxBitmap to wxImage conversion (wxMSW).

a-wi opened the pull request:
Added ability to convert 32bpp wxBitmap explicitly marked as ARGB to wxImage with alpha channel values.
See http://trac.wxwidgets.org/ticket/14582

vadz commented on include/wx/msw/dib.h:
```diff
@@ -164,7 +164,23 @@ class WXDLLIMPEXP_CORE wxDIB
+        // wxImage with alpha channel values in this case.
+        Convert_AlphaAlwaysIf32bpp
+    };
+#if WXWIN_COMPATIBILITY_3_0
```
Is there any problem with using `ConvertToImage(ConversionFlags flags = Convert_AlphaAuto)` always? I don't see how is it incompatible with the 3.0 API... Well, actually I do, it would break the existing code taking a pointer to a member function, but this seems so unlikely and is easy to fix (because it would be a compilation error), that I don't think it's really a problem.

So I'd only keep the `#else` branch here.

vadz commented on src/msw/bitmap.cpp:
```diff
@@ -932,7 +932,13 @@ wxImage wxBitmap::ConvertToImage() const
 
     // and then DIB to our wxImage
-    wxImage image = dib.ConvertToImage();
+    // Enforce creating wxImage with alpha values
```
I'd write something like this here:

    // By default, we autodetect the presence of alpha and consider unnecessary to create
    // the alpha channel in wxImage if we don't have any effective alpha in the bitmap because
    // its depth, on its own, is not an indicator that it uses alpha as it could be just the default
    // screen depth. However if the user had explicitly called UseAlpha(), then we consider
    // that the resulting image should really have the alpha channel too.

vadz commented:
Thanks, I agree with this approach, but I think we could simplify things by not worrying about 3.0 compatibility so much -- unless I'm missing some more important compatibility breakage than the one I identified (taking a pointer to member).

a-wi committed:
Add method to convert wxDIB to wxImage enforcing alpha channel values (wxMSW).

wxDIB::ConvertToImage called with Convert_AlphaAuto converts wxDIB to wxImage with automatic checking if 32 bpp DIB contains real alpha values (legacy way).
When it is called with Convert_AlphaAlwaysIf32bpp then automatic checking is disabled and 32 bpp DIB is unconditionally converted to ARGB wxImage.

a-wi committed:
Convert 32bpp wxBitmap with alpha flag set to wxImage with alpha channel values.

Create wxImage with alpha values if bitmap is explicitly marked as an ARGB one. Otherwise it will be checked automatically whether 32bpp bitmap contains real alpha channel values or not and output wxImage will be created with or without alpha channel values.

a-wi commented:
Commits updated.

tierra commented:
I just retriggered that AppVeyor build since it took too long to run (it auto-kills the build after 60 minutes). It's usually unrelated to your changes.
